pkg/middleware/stored_session.go

```diff
@@ -128,6 +197,11 @@ func (s *storedSessionLoader) refreshSessionIfNeeded(rw http.ResponseWriter, req
 	return s.validateSession(req.Context(), session)
 }
 
+// needsRefresh determines whether we should attempt to refresh a session or not.
+func needsRefresh(refreshPeriod time.Duration, session *sessionsapi.SessionState) bool {
+	return refreshPeriod > time.Duration(0) && session.Age() > refreshPeriod
```

Comment:
is the first part `refreshPeriod > time.Duration(0)` a sanity check to make sure that refresh period is not zero or something?

Reply:
Yeah that's it, we use the 0 value as a "disable refresh", so if the refresh period is 0, we never even attempt a refresh. Some users may not want us to attempt a refresh so we had to have a way to disable it